docs/source/notebooks_and_ipython/add_kedro_to_a_notebook.md

```diff
@@ -0,0 +1,652 @@
+
+# Load the datasets
+...
+```
+<!-- #endregion -->
+
+It's also possible to use the configuration loader to initialise the Data Catalog. As we saw from loading parameter values, there is some initial setup because of the Kedro project structure:
+
+1. If you haven't already, in your notebook folder, create a subfolder called `conf`. 
+2. Within the `conf` subfolder, create a subfolder called `base`. 
+3. Copy your `catalog.yml` into `conf/base/` so you have a file with the path `conf/base/catalog.yml`.
+4. Within the `conf` subfolder, if you haven't already, create a subfolder called `local`. 
+
+After this setup, you can use Kedro's `OmegaConfigLoader` in code as follows:
+
+```python
+# Now we are using Kedro's ConfigLoader alongside the DataCatalog
+
+from kedro.config import OmegaConfigLoader
+from kedro.framework.project import settings
+from kedro.io import DataCatalog
+
+conf_loader = OmegaConfigLoader("conf")
+conf_catalog=conf_loader["catalog"]
+
+# Create the DataCatalog instance from the configuration
+catalog = DataCatalog.from_config(conf_catalog)
+
+# Load the datasets
+companies = catalog.load("companies")
+reviews = catalog.load("reviews")
+shuttles = catalog.load("shuttles")
+```
+
+## Where next?
+At this point in the notebook, we've introduced Kedro data management (using the Data Catalog) and configuration loader. TO DO -- ADD SOME REASONING FOR WHY KEDRO ENCOURAGES THIS AND LINK TO BEST PRACTICES. You have now "Kedro-ised" the notebook code to make it more reusable in future. However, you can go further if your ultimate goal is to migrate code out of the notebook and use it in a full-blown Kedro project. 
+
+## Refactor your code into functions
+Code in a Kedro project runs in one or more pipelines, where a pipeline is a series of nodes, which wrap discrete functions. If your ultimate goal is to create a Kedro project and use it to run your notebook code, you'll need to take the code that processes the data, and the code that calls the data science model, and make it into one or more nodes. 
+
+One option is to simply put everything into a single function. Let's try this.
+
+```python
+# Use Kedro for data management and configuration
+
+from kedro.config import OmegaConfigLoader
+from kedro.framework.project import settings
+from kedro.io import DataCatalog
+
+conf_loader = OmegaConfigLoader("conf")
+conf_catalog=conf_loader["catalog"]
+conf_params=conf_loader["parameters"]
+
+# Create the DataCatalog instance from the configuration
+catalog = DataCatalog.from_config(conf_catalog)
+
+# Load the datasets
+companies = catalog.load("companies")
+reviews = catalog.load("reviews")
+shuttles = catalog.load("shuttles")
+
+# Load the configuration data
+test_size = conf_params["model_options"]["test_size"]
+random_state = conf_params["model_options"]["random_state"]
+
+```
+
+```python
+def big_function():
+    
+    ####################
+    # Data processing  #
+    ####################
```

Comment:
I like this idea. but I wouldn't even include the full code there, I'd do something like

```python
def big_function():
    ...

big_function()
```

(yes, `...` is valid Python syntax) and skip to the next part. It's mostly for narrative purposes, we don't want users to even write this I think.

Reply:
This is the only change I've yet to make but I wanted to discuss it because I think it's quite common to do. I've seen a conversation with @noklam somewhere on Slack recently where someone was saying that was the common first step in Kedro-ising for them -- make one big func and then refactor. So I thought it worth including. I don't know though and am happy to be overruled 🤷

Reply:
Fair enough, not the hill I want to die on 😄

Reply:
Yay, `big_function()` lives to fight another day ⚔️